src/app/containers/SettingsPage/index.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+// APIs
+import { apiGetMyUserInfo, apiUpdateUserInfo } from 'src/api/user';
+
+// HOC
+import { withRouter } from 'react-router-dom';
+import withUserLogin from 'src/shared/hoc/withUserLogin';
+import { UserInfoContext } from 'src/shared/context/UserInfo';
+
+// Components
+import Loading from '../../common-ui/Loading';
+
+// UI
+import * as MainPanel from '../../common-ui/MainPanel';
+import * as Form from '../../common-ui/Form';
+import { PrimaryButton } from '../../common-ui/Button';
+import { Helmet } from 'react-helmet';
+import { ErrorPopup, SuccessPopup } from '../../common-ui/Popup';
+
+// Constants and utils
+import { API_PROGRESS } from 'src/shared/constants';
+import { validate } from './validators';
+
+const labels = {
+  fullname: 'Họ và tên',
+  school: 'Trường học',
+  email: 'Email liên lạc',
+};
+
+function SettingsPage({ history, location }) {
+  const { userInfo } = React.useContext(UserInfoContext);
+  const { full_name: fullname, school_name: school, email } = apiGetMyUserInfo(userInfo.token);
+  const [values, setValues] = React.useState({
+    fullname,
+    school,
+    email,
+  });
+  const [errors, setErrors] = React.useState({});
+  const [hasError, setHasError] = React.useState(false);
+  const [apiState, setApiState] = React.useState({
+    progress: API_PROGRESS.INIT,
+    code: null,
+    msg: null,
+  });
+
+  const handleChange = (name, value) => {
+    setValues({ ...values, [name]: value });
+    setErrors({ ...errors, [name]: null });
+  };
+
+  const defaultProps = (name) => ({
+    label: labels[name],
+    name,
+    value: values[name],
+    onChange: (newValue) => handleChange(name, newValue),
+    error: errors[name],
+  });
+
+  const handleSubmit = React.useCallback(
+    async (event) => {
+      event.preventDefault();
+      const validation = validate(values);
+      setValues(validation.newValues);
+      setErrors(validation.errors);
+
+      if (validation.hasError) {
+        setHasError(true);
+        return;
+      }
+
+      setApiState({ progress: API_PROGRESS.REQ, code: null, msg: null });
+      const { code, data, msg } = await apiUpdateUserInfo({ fullname, school });
+
+      if (code || !data) {
+        setApiState({ progress: API_PROGRESS.FAILED, code, msg });
+      } else {
+        setApiState({ ...apiState, progress: API_PROGRESS.SUCCESS });
+      }
+    },
+    [values],
+  );
+
+  return (
+    <MainPanel.Container>
+      <Helmet>
+        <title>Cài đặt</title>
+      </Helmet>
+      {apiState.progress === API_PROGRESS.REQ ? (
+        <Loading />
+      ) : hasError ? (
+        <ErrorPopup
+          show
+          content="Thông tin nhập không hợp lệ!"
+          onClose={() => setHasError(false)}
+        />
```

Comment:
this is front-end validation error, we do not need to show popup for such error, can use the inline error. No need to render such popup.